x-pack/plugins/reporting/server/routes/common/generate/request_handler.ts

```diff
@@ -80,14 +87,18 @@ export class RequestHandler {
     jobParams.version = checkParamsVersion(jobParams, logger);
 
     // 2. Encrypt request headers to store for the running report job to authenticate itself with Kibana
-    const headers = await this.encryptHeaders();
+    // IMPORTANT: We need to store `usesDedicatedApiKey` flag in the report so we can determine if we need to invalidate
```

Comment:
>[!NOTE]
> I’ve implemented `disposeReportHeaders` function, which requires both the `usesDedicatedApiKey` flag and the decrypted headers, but I couldn’t quickly find a place where we can call it for all reports. I assume we need to store the flag in the report payload somehow to reuse it later.

Reply:
`disposeReportHeaders` should go into the `run` method of the `report:execute` task declaration: https://github.com/elastic/kibana/blob/f1440787129c973fdae0a8fa419f8909029601ae/x-pack/plugins/reporting/server/lib/tasks/execute_report.ts#L531